src/wms/src/main/java/org/geoserver/wms/legendgraphic/FeatureCountProcessor.java

```diff
@@ -280,8 +281,16 @@ private GetMapRequest parseAssociatedGetMap(LegendRequest legend, Rule[] rules)
         rawKvp.put("STYLES", "");
         // ... width and height
         rawKvp.put("WIDTH", rawKvp.get("SRCWIDTH"));
-        rawKvp.put("HEIGTH", rawKvp.get("SRCHEIGHT"));
-        
+        rawKvp.put("HEIGHT", rawKvp.get("SRCHEIGHT"));
+        // ... set default values if not yet set
+        rawKvp.putIfAbsent("HEIGHT", String.valueOf(GetLegendGraphicRequest.DEFAULT_HEIGHT));
```

Comment:
Having a value of "20" is unreasonable, most maps with scale dependencies will simply not show at that scale. Also, the sizing is not proportional to the map own bounding box, creating a different scale denominator in south/north east/west directions. See https://github.com/geoserver/geoserver/blob/master/src/wms/src/main/java/org/geoserver/wms/DefaultWebMapService.java#L445 (maybe it's possible to just call that method, something to try out)